lib/composer.js

```diff
@@ -216,38 +216,39 @@ export default class Composer extends Disposable {
     }
     this.cachedDiCy.set(filePath, dicy)
 
-    dicy.on('log', event => {
-      const nameText = event.name ? `[${event.name}] ` : ''
-      const typeText = event.category ? `${event.category}: ` : ''
-      const message = {
-        type: event.severity,
-        text: `${nameText}${typeText}${event.text}`
-      }
+    dicy
+      .on('log', event => {
+        const nameText = event.name ? `[${event.name}] ` : ''
+        const typeText = event.category ? `${event.category}: ` : ''
+        const message = {
+          type: event.severity,
+          text: `${nameText}${typeText}${event.text}`
+        }
 
-      if (event.source) {
-        message.filePath = path.resolve(dicy.rootPath, event.source.file)
-        if (event.source.range) {
-          message.range = [[event.source.range.start - 1, 0], [event.source.range.end - 1, Number.MAX_SAFE_INTEGER]]
+        if (event.source) {
+          message.filePath = path.resolve(dicy.rootPath, event.source.file)
+          if (event.source.range) {
+            message.range = [[event.source.range.start - 1, 0], [event.source.range.end - 1, Number.MAX_SAFE_INTEGER]]
+          }
         }
-      }
 
-      if (event.log) {
-        message.logPath = path.resolve(dicy.rootPath, event.log.file)
-        if (event.log.range) {
-          message.logRange = [[event.log.range.start - 1, 0], [event.log.range.end - 1, Number.MAX_SAFE_INTEGER]]
+        if (event.log) {
+          message.logPath = path.resolve(dicy.rootPath, event.log.file)
+          if (event.log.range) {
+            message.logRange = [[event.log.range.start - 1, 0], [event.log.range.end - 1, Number.MAX_SAFE_INTEGER]]
+          }
         }
-      }
 
-      latex.log.showMessage(message)
-    })
-    .on('command', event => {
-      latex.log.info(`[${event.rule}] Executing \`${event.command}\``)
-    })
-    .on('fileDeleted', event => {
-      if (!event.virtual) {
-        latex.log.info(`Deleting \`${event.file}\``)
-      }
-    })
+        latex.log.showMessage(message)
+      })
+      .on('command', event => {
+        latex.log.info(`[${event.rule}] Executing \`${event.command}\``)
+      })
+      .on('fileDeleted', event => {
+        if (!event.virtual) {
+          latex.log.info(`Deleting \`${event.file}\``)
+        }
+      })
```

Comment:
I'd prefer it if all of these `.on` calls were not chained like this. Right now the code is pretty difficult to read. Each `.on` call should a separate `dicy.on` call, perhaps with a blank line separating each of them.

Reply:
Sounds good.